docs/content/concepts/configuration/config-schema.mdx

```diff
@@ -117,167 +145,42 @@ dagster job execute --config my_config.yaml
-
-```python file=/concepts/configuration/make_values_resource_run_config.py startafter=start_run_config_1 endbefore=end_run_config_1
-result = file_dir_job.execute_in_process(
-    run_config={"resources": {"file_dir": {"config": "/my_files/"}}}
+    [greeting],
+    run_config=RunConfig({"greeting": MyAssetConfig(nonexistent_config_param=1)}),
 )
 ```
 
-Alternatively, if you want to provide different config values for each op within a job, you can also specify the schema of the values like:
-
-```python file=/concepts/configuration/make_values_resource_config_schema.py startafter=start_file_example endbefore=end_file_example
-@op(required_resource_keys={"file_dirs"})
-def write_file(context):
-    filename = f"{context.resources.file_dirs['write_file_dir']}/new_file.txt"
-    open(filename, "x", encoding="utf8").close()
-
-    context.log.info(f"Created file: {filename}")
+### Using environment variables with config
 
+Ops and assets can be configured using environment variables by passing an <PyObject object="EnvVar" /> when constructing your config object. This is useful when the value may vary based on environment or is sensitive. If you're using Dagster Cloud, environment variables can be [set up directly in the UI](/dagster-cloud/developing-testing/environment-variables-and-secrets).
 
-@op(required_resource_keys={"file_dirs"})
-def total_num_files(context):
-    files_in_dir = os.listdir(context.resources.file_dirs["count_file_dir"])
-    context.log.info(f"Total number of files: {len(files_in_dir)}")
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_execute_with_config_envvar endbefore=end_execute_with_config_envvar dedent=4
+from dagster import job, materialize, op, RunConfig, EnvVar
 
-
-@job(
-    resource_defs={
-        "file_dirs": make_values_resource(write_file_dir=str, count_file_dir=str)
-    }
+job_result = greeting_job.execute_in_process(
+    run_config=RunConfig({"print_greeting": MyOpConfig(person_name=EnvVar("PERSON_NAME"))})
 )
-def file_dirs_job():
-    write_file()
-    total_num_files()
-```
 
-You can specify run config like so:
-
-```python file=/concepts/configuration/make_values_resource_run_config.py startafter=start_run_config_2 endbefore=end_run_config_2
-result = file_dirs_job.execute_in_process(
-    run_config={
-        "resources": {
-            "file_dirs": {
-                "config": {
-                    "write_file_dir": "/write_files/",
-                    "count_file_dir": "/count_files/",
-                }
-            }
-        }
-    }
+asset_result = materialize(
+    [greeting],
+    run_config=RunConfig(
+        {"greeting": MyAssetConfig(person_name=EnvVar("PERSON_NAME"))}
+    ),
 )
 ```
 
-Or pass these values via a YAML file:
-
-```YAML file=/concepts/configuration/make_values_resource_values.yaml
-resources:
-  file_dirs:
-    config:
-      write_file_dir: /write_files/
-      count_file_dir: /count_files/
-```
-
 ---
 
 ## See it in action
```

Comment:
Will the example linked in this section still be applicable, given the changes to config?

Reply:
Not until it's cut over - I will remove the external link for now and we can re-add once it's updated.